### ISSUE TITLE
cdktf does not work with 3rd party providers

### ISSUE THREAD

**Issue by trondhindenes** [hashicorp/terraform-cdk#607]
NOTE: I really wish the issue template was easier to use. I want to provide some background and info, and there's no natural place for me in this template to do that.

### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions
```
╰─ cdktf --version
0.2.0
```

### Affected Resource(s)
I don't know what that you're asking for here. My bug isn't related to a specific resource

I'll just write what I'm seeing, sorry for messing up the issue template:

I'm trying to use cdktf with the `datadog` provider. As far as I understand, this is not a hashcorp-owned provider, so its proper "id" is `datadog/datadog`. The CDK has an example for using the `dnsimple` provider which is also 3rd party, and its id is `dnsimple/dnsimple.

So: If I just add "datadog" to my cdktf.json file, like so:
```
  "terraformProviders": [
    "random@~> 3.1.0",
    "datadog"
  ],
```
`cdktf get` will produce an error:
```
Could not retrieve the list of available versions for provider
hashicorp/datadog: provider registry registry.terraform.io does not have a
provider named registry.terraform.io/hashicorp/datadog
```
So it's pretty clear to me that the documentation should be updated to reflect this, the correct way of specifying 3rd party providers is by using the full name, such as `datadog/datadog` (or `dnsimple/dnsimple`). If I correct this, then `cdktf get` will successfully download/generate the `datadog` provider, and make it importable in my code.

However:
When executing using either `cdktf deploy` or cd-ing int the "cdktf.out" dir and run `terraform init`, I'm getting a similar error:
```
╰─ terraform init

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/datadog...
- Finding hashicorp/google versions matching "~> 3.0"...
- Installing hashicorp/google v3.60.0...
- Installed hashicorp/google v3.60.0 (signed by HashiCorp)
```
As you can see, something in cdktf strips away the "owner" part of the provider, making terraform assume it's a hashcorp-owned module. This causes the error:

```
Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider
hashicorp/datadog: provider registry registry.terraform.io does not have a
provider named registry.terraform.io/hashicorp/datadog

If you have just upgraded directly from Terraform v0.12 to Terraform v0.14
then please upgrade to Terraform v0.13 first and follow the upgrade guide for
that release, which might help you address this problem.

Did you intend to use datadog/datadog? If so, you must specify that source
address in each module which requires that provider. To see which modules are
currently depending on hashicorp/datadog, run the following command:
    terraform providers

```

Bottom line: I'm unable to use cdktf with providers not supplied by hashicorp.

**Comment by trondhindenes** [comment]
Wondering if this is related: https://github.com/hashicorp/terraform-cdk/issues/118

**Comment by trondhindenes** [comment]
might be worth mentioning that the `required_providers` block of the `cdk.tf.json` file procuced by `cdktf synth` does not include the datadog provider. Not sure why.

**Comment by jsteinich** [comment]
@trondhindenes were you able to get this working correctly? Looking at the [Terraform provider documentation](https://registry.terraform.io/providers/DataDog/datadog/latest/docs), it looks like `DataDog/datadog` is the id to be used.

**Comment by trondhindenes** [comment]
I was. I do believe the docs have to be updated to use the "fully qualified" name for third party providers, but the rest was a code bug that was embarrassing enough for me to just want to forget about it :-)
